Modules/_collectionsmodule.c

```diff
@@ -413,10 +420,10 @@ Add an element to the left side of the deque.
 [clinic start generated code]*/
 
 static PyObject *
-deque_appendleft(dequeobject *deque, PyObject *item)
-/*[clinic end generated code: output=de0335a64800ffd8 input=bbdaa60a3e956062]*/
+deque_appendleft_impl(dequeobject *deque, PyObject *item)
+/*[clinic end generated code: output=9a192edbcd0f20db input=236c2fbceaf08e14]*/
 {
-    if (deque_appendleft_internal(deque, Py_NewRef(item), deque->maxlen) < 0)
+    if (deque_appendleft_lock_held(deque, Py_NewRef(item), deque->maxlen) < 0)
```

Comment:
I'm a little bit puzzled by this naming; initially, I read it as "append left with lock held", but I see it is the opposite: "append left, no lock held". Unless this is already the established naming convention, could we consider a more unambiguous naming scheme?

Reply:
> Unless this is already the established naming convention, could we consider a more unambiguous naming scheme?

This is something that we bikeshed about internally at Meta (before folks had access to Discord) for quite a while before landing on the `_lock_held` suffix. Ironically, folks felt it was the least ambiguous of the suffixes that were discussed. I think a few PRs have since landed that use this suffix, so while it's not established, it's on its way there. I don't think anyone is particularly enamored with this suffix; if you have an alternative that you think is better, please say so.

> OTOH, if this is the convention, it should be documented somewhere (possibly the devguide).

Agreed.

Reply:
> I think a few PRs have since landed that use this suffix, so while it's not established, it's on its way there. 

Ok, I don't think it is worth the churn to change it. Documenting it sounds like the best way forward.